composer.json

```diff
@@ -31,7 +31,7 @@
 		"ext-libxml": "*",
 		"ext-mbstring": "*",
 		"ext-simplexml": "*",
-		"league/flysystem": "^1.0",
+		"league/flysystem": "^2.0 || ^3.0",
```

Comment:
i guess it should considered as breaking change

Reply:
Not sure it possible in that repo

Reply:
It is possible for the main branch on which you're doing this PR, but I don't think the major bump is necessary, as the Flysystem change doesn't impact anything outside, all the changes are contained within this library.
If any doubt, proceed with major bump, but maybe check before if this impacts conflicts with the version of Flysystem optionally used in the code using this library first.

Reply:
yes, it influent for two places and it was corrected also
https://github.com/oat-sa/generis/pull/1028
https://github.com/oat-sa/lib-flysystem-filecache/pull/8

Reply:
will i be able update tao-deliver-be on that version with flysytem 1 installed?
I guess no
<img width="1289" alt="Screenshot 2022-11-25 at 14 37 41" src="https://user-images.githubusercontent.com/37804293/203997189-141d7e6b-f9b9-4f10-843b-922998a8f1f8.png">

it require from me additioanal adjustment (update flysystem) to be able update this library?
isn't it should be considered as breaking change?

Reply:
I agree that this is a breaking change. Not because of the dependency, though, but because the public contract actually changes.
Take `public function setFilesystem(Filesystem $filesystem = null)` method. Yes, it still accepts the same `Filesystem` instance, but the interface behind it is totally different.